tests/utils/volumesnapshot.go

```diff
@@ -0,0 +1,40 @@
+package utils
+
+import (
+	"context"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	snapshotv1 "github.com/kubernetes-csi/external-snapshotter/client/v6/apis/volumesnapshot/v1"
+	"github.com/onsi/gomega"
+)
+
+// NewVolumeSnapshot initializes a VolumeSnapshot struct
+func NewVolumeSnapshot(name, namespace, sourcePvcName string, snapshotClassName *string) *snapshotv1.VolumeSnapshot {
+	return &snapshotv1.VolumeSnapshot{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: snapshotv1.VolumeSnapshotSpec{
+			Source: snapshotv1.VolumeSnapshotSource{
+				PersistentVolumeClaimName: &sourcePvcName,
+			},
+			VolumeSnapshotClassName: snapshotClassName,
+		},
+	}
+}
+
+// WaitSnapshotReady waits until the snapshot is ready to be used
```

Comment:
In all your calls to this func you are Get()ting again the VolumeSnapshot, so why not return it here?

Reply:
I think it may be a little unexpected for this util to do something other than return the struct

Reply:
so why not pass `snapshot` instead of `snap` to `Get()` save the additional `Get()` after this func is called?

Reply:
ohh you are talking about `WaitSnapshotReady`, thought you meant `NewVolumeSnapshot`
I think mutating the pointer is a little weird, maybe have this func return latest snap state?

Reply:
We don't mutate the pointer, we just referesh it's refered CR state by Get(..., snapshot). The line above it will sure be deleted.